### PR TITLE
[ MB-16199 ] Ensure service items have an origin and final destination address

### PR DESCRIPTION
**Pull request by rogeruiz**
- Refactoring NewMTOServiceItemUpdater;
- Making the function signatures more legible;

## Summary

The goal of this patch is to have `DDDSIT` service items create a destination
origin address. This will allow for the address change calculations introduced
in #10649 and will help unblock other user stories as well.

## Verification Steps for the Author

These are to be checked by the author.

- [ ] Have the Jira acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/run-storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/about/application-setup/milmove-local-client/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/about/development/testing)

### How to test

_More information on how to properly test this manually coming soon..._

### CODE REVIEW

**Comment by rogeruiz** [comment]
Currently, this does not include any changes around adding the SIT Final Destination Address in the Create MTO Service Item request. We will need to add this if tomorrow's meeting determines we should.

File: pkg/services/mto_service_item/mto_service_item_updater.go [hunk 1]
```diff
@@ -113,6 +131,28 @@ func (p *mtoServiceItemUpdater) updateServiceItem(appCtx appcontext.AppContext,
...
+				serviceItem.SITDestinationOriginalAddress = serviceItem.SITDestinationFinalAddress
+			} else {
+				// TODO: Do we need to check for this to have an error?
+				mtoShipment, _ := p.shipmentFetcher.GetShipment(appCtx, *serviceItem.MTOShipmentID, "DestinationAddress")
+				// Set the original address on a service item to the shipment's
+				// destination address when approving a SIT service item.
+				serviceItem.SITDestinationOriginalAddressID = mtoShipment.DestinationAddressID
+				serviceItem.SITDestinationOriginalAddress = mtoShipment.DestinationAddress
+			}
+		}
```
Comment (ronaktruss) [inline pkg/services/mto_service_item/mto_service_item_updater.go:155]: So far I have just read the code and most of it looks good at a high level. You may run into an issue where the serviceItem.SITDestinationFinalAddress is always nil. The serviceItem is being fetched and defined by the `findServiceItem` function in this file which doesn't eagerPreload the address information.

Comment (rogeruiz) [reply pkg/services/mto_service_item/mto_service_item_updater.go:155]: ah thank you! I'll refactor this.

**Review by michaelvaldes-truss — CHANGES_REQUESTED** [review]
This looks great so far, and it works on my machine 🎉 

**One additional suggestion:** can we update the scenarios that create approved DDDSIT service items in pkg/testdatagen/testharness/make_move.go to include the required Original Address?

File: pkg/services/mto_service_item/mto_service_item_updater.go [hunk 2]
```diff
@@ -50,7 +61,10 @@ func (p *mtoServiceItemUpdater) ApproveOrRejectServiceItem(appCtx appcontext.App
 
 func (p *mtoServiceItemUpdater) findServiceItem(appCtx appcontext.AppContext, serviceItemID uuid.UUID) (*models.MTOServiceItem, error) {
 	var serviceItem models.MTOServiceItem
-	err := appCtx.DB().Q().EagerPreload("MoveTaskOrder").Find(&serviceItem, serviceItemID)
+	err := appCtx.DB().Q().EagerPreload(
+		"MoveTaskOrder",
+		"MTOServiceItems.SITDestinationFinalAddress",
```
Comment (donaldthai) [inline pkg/services/mto_service_item/mto_service_item_updater.go:66]: Issue: Trying to approve/reject the SIT service item and I'm getting an error from this eager preload string. We typically just need the property name to load it in the struct.

Suggestion: I think we could just remove the `MTOServiceItems.` part to resolve the issue.

**Review by donaldthai — CHANGES_REQUESTED** [review]
Looking good so far. Added some comments!

File: pkg/services/mto_service_item/mto_service_item_updater.go (hunk 2 above)
Comment (rogeruiz) [inline pkg/services/mto_service_item/mto_service_item_updater.go:66]:
```suggestion
		"SITDestinationFinalAddress",
```

**Comment by rogeruiz** [comment]
Okay, I've refactored based on the feedback I received here. Thanks folks.

**Comment by donaldthai** [comment]
@Mergifyio update

**Comment by ronaktruss** [comment]
Functionally this looks good to me. I tested the scenario where the prime provides a final destination address and when they don't. I didn't test the scenario where there isn't a shipment destination (I think there's another ticket to populate this).

**Comment by donaldthai** [comment]
@Mergifyio update

**Review by michaelvaldes-truss — APPROVED** [review]
Lgtm -- approving a destination SIT ensures that the Original Address field is populated appropriately.
I don't want to delay this PR any longer since it's currently blocking acceptance testing, but can we update the scenarios that create approved DDDSIT service items in pkg/testdatagen/testharness/make_move.go to include the required Original Address?
This would ensure that our testharness scenarios are functional for SIT address update testing.

**Comment by rogeruiz** [comment]
I'll update our test scenarios after this gets merged in on a separate PR.

**Comment by rogeruiz** [comment]
https://github.com/transcom/mymove/pull/10880

**Pull request merged**